my-webxr-app/src/repository/DbRepository.tsx

```diff
@@ -0,0 +1,137 @@
+        @param column: the column to be added to the database
+        @return Promise<string>: the primary key of the column aka the name of the column
+         */
+  async addColumn(column: Column) {
+    return this.columns.add(column);
+  }
+
+  /*
+        getColumn gets a column from the database
+        @param columnName: the name of the column to be retrieved
+        @return Promise<Column>
+         */
+  private async getColumn(columnName: string) {
+    const column = await this.columns
+      .where('name')
+      .equals(columnName)
+      .toArray();
+    assert.ok(column.length <= 1, `Found more than one column with name ${columnName}!`);
+    assert.ok(column.length === 1, `Column ${columnName} does not exist!`);
+
+    return column[0];
+  }
+
+  /*
+        getPoints gets the points from the database
+        @param qualifyingPointOnly: if true, only return points that have no missing data
+        @param columnXName: the name of the column to be used as the x-axis
+        @param columnYName: the name of the column to be used as the y-axis
+        @param columnZName: the name of the column to be used as the z-axis
+        @pre-condition: 3 column names must be distinct and existing in the db
+        @return Promise<Array<DataPoint>>
+         */
+  async getPoints(
+    qualifyingPointOnly: boolean,
+    columnXName: string,
+    columnYName: string,
+    columnZName: string,
+  ): Promise<Array<DataPoint>> {
+    // verify the three columns are distinct
+    assert.equal(
+      (new Set([columnXName, columnYName, columnZName])).size,
+      3,
+      `The three columns must be distinct but got: ${columnXName},${
+        columnYName},${columnZName}!`,
+    );
+
+    // get the three columns
+    const columnX = await this.getColumn(columnXName);
+    const columnY = await this.getColumn(columnYName);
+    const columnZ = await this.getColumn(columnZName);
+
+    const sameLength = new Set([columnX.values.length,
+      columnY.values.length,
+      columnZ.values.length]);
+    assert.equal(sameLength.size, 1, 'The number of values in the given columns must be the same, but '
+            + `column ${columnXName} has ${columnX.values.length} values, `
+            + `column ${columnYName} has ${columnY.values.length} values, and `
+            + `column ${columnZName} has ${columnZ.values.length} values!`);
+
+    const dataPoints = DbRepository.convertColumnsIntoDataPoints(
+      qualifyingPointOnly,
+      columnX,
+      columnY,
+      columnZ,
+    );
+    return Promise.resolve(dataPoints);
+  }
+
+  /*
+        convertColumnsIntoDataPoints converts the columns into an array of DataPoints
+        @param qualifyingPointOnly: if true, only return points that have no missing data
+        @param columnX: the column to be used as the x-axis
+        @param columnY: the column to be used as the y-axis
+        @param columnZ: the column to be used as the z-axis
+        @return Array<DataPoint>
+         */
```

Comment:
Are these supposed to be indented like this?

Reply:
^

Reply:
yeah, it was done automatically with 'npm run lint' command!

Reply:
ok, you can disregard my comment then with the approval, figured it was prob the linter

Reply:
I'm thinking the linter was looking for
```ts
/*
 * My docstring
 */
```
rather than what you have here (just missing the extra `*` on each line).